src/coreclr/vm/ecall.cpp

```diff
@@ -186,6 +186,54 @@ void ECall::PopulateManagedHelpers()
     pMD = CoreLibBinder::GetMethod((BinderMethodID)(METHOD__MATH__CONVERT_TO_UINT64_CHECKED));
     pDest = pMD->GetMultiCallableAddrOfCode();
     SetJitHelperFunction(CORINFO_HELP_DBL2ULNG_OVF, pDest);
+
+    pMD = CoreLibBinder::GetMethod((BinderMethodID)(METHOD__THROW_HELPERS__THROWARGUMENTEXCEPTION));
```

Comment:
To your earlier question about this being executed during startup for no good reason, it should be a straightforward change to do this lazily only once somebody asks for the address of the JIT helper (typical app will never ask for address for most of these throwing helpers):

- There are the 3 places in jitinterface.cpp that needs to be updated to call the lazy initialization:

            Precode* pPrecode = Precode::GetPrecodeFromEntryPoint((PCODE)hlpDynamicFuncTable[dynamicFtnNum].pfnHelper);

        pfnHelper = hlpDynamicFuncTable[dynamicFtnNum].pfnHelper;

        pfnHelper = hlpDynamicFuncTable[((size_t)pfnHelper - 1)].pfnHelper;

- The lazy initialization would convert CorInfoHelpFunc to BinderMethodID, get the helper entrypoint (`CoreLibBinder::GetMethod`, `GetMultiCallableAddrOfCode`) and cache the pointer in `hlpDynamicFuncTable`

Reply:
Done. This increased the PR size a bit, but it does seem to remove some cruft and reduce any concern I had with start-up impact.